### PR TITLE
ComicK: Add a new tag based on "Origination"

### LINKED ISSUE #17085: Comick: Add new tag based on "Origination"
### Source name

Comick

### Source language

English

### Describe your suggested feature

Origination: Manhwa
image

Add new tag "Manhwa"
image

*This will make it easier to filter manga/manhwa/manhua*

### Other details

_No response_

### Acknowledgements

- [X] I have searched the existing issues and this is a new ticket, **NOT** a duplicate or related to another open or closed issue.
- [X] I have written a short but informative title.
- [X] If this is an issue with the app itself, I should be opening an issue in the [app repository](https://github.com/tachiyomiorg/tachiyomi/issues/new/choose).
- [X] I have updated the app to version **[0.14.6](https://github.com/tachiyomiorg/tachiyomi/releases/latest)**.
- [X] I will fill out all of the requested information in this form.

### PR DESCRIPTION
**Pull request by iD-666**
Closes #17085

I hope someone could review this and see if it works.

Checklist:

- [x] Updated `extVersionCode` value in `build.gradle` for individual extensions
- [ ] Updated `overrideVersionCode` or `baseVersionCode` as needed for all multisrc extensions
- [x] Referenced all related issues in the PR body (e.g. "Closes #xyz")
- [ ] Added the `isNsfw = true` flag in `build.gradle` when appropriate
- [x] Have not changed source names
- [ ] Have explicitly kept the `id` if a source's name or language were changed
- [ ] Have tested the modifications by compiling and running the extension through Android Studio

### CODE REVIEW

File: src/all/comickfun/src/eu/kanade/tachiyomi/extension/all/comickfun/ComickFunDto.kt [hunk 1]
```diff
@@ -26,7 +26,7 @@ data class Manga(
     val comic: Comic,
     val artists: List<Artist> = emptyList(),
     val authors: List<Author> = emptyList(),
-    val genres: List<Genre> = emptyList(),
+    val genres: List<Genre> = listOf(comic.comicType()?.name),
```
Comment (AwkwardPeak7) [inline src/all/comickfun/src/eu/kanade/tachiyomi/extension/all/comickfun/ComickFunDto.kt:29]: This won't work as you are providing the comic type as default value, which upon deserialization will use data in the json response.

File: src/all/comickfun/src/eu/kanade/tachiyomi/extension/all/comickfun/ComickFunDto.kt [hunk 2]
```diff
@@ -48,21 +48,31 @@ data class Manga(
         thumbnail_url = parseCover(comic.cover_url, comic.md_covers)
         artist = artists.joinToString { it.name.trim() }
         author = authors.joinToString { it.name.trim() }
-        genre = genres.joinToString { it.name.trim() }
+        genre = genres.filterNotNull().joinToString { it.name.trim() }
```
Comment (AwkwardPeak7) [inline src/all/comickfun/src/eu/kanade/tachiyomi/extension/all/comickfun/ComickFunDto.kt:51]: append the comic type to genre string here.

**Review by AwkwardPeak7 — COMMENTED** [review]
Please do test your changes.

File: src/all/comickfun/src/eu/kanade/tachiyomi/extension/all/comickfun/ComickFunDto.kt [hunk 3]
```diff
@@ -48,21 +48,31 @@ data class Manga(
...
-)
+) {
+    fun comicType(): Genre? {
+        return when (country) {
+            "jp" -> Genre("manga", "Manga")
+            "kr" -> Genre("manhwa", "Manhwa")
+            "cn" -> Genre("manhua", "Manhua")
+            else -> null
+        }
+    }
```
Comment (AntsyLich) [inline src/all/comickfun/src/eu/kanade/tachiyomi/extension/all/comickfun/ComickFunDto.kt:74]:
```suggestion
    val typeGenre: Genre? = when (country) {
        "jp" -> Genre("manga", "Manga")
        "kr" -> Genre("manhwa", "Manhwa")
        "cn" -> Genre("manhua", "Manhua")
        else -> null
    }
```

File: src/all/comickfun/src/eu/kanade/tachiyomi/extension/all/comickfun/ComickFunDto.kt (hunk 1 above)
Comment (AntsyLich) [inline src/all/comickfun/src/eu/kanade/tachiyomi/extension/all/comickfun/ComickFunDto.kt:29]:
```suggestion
    val genres: List<Genre> = emptyList(),
```

File: src/all/comickfun/src/eu/kanade/tachiyomi/extension/all/comickfun/ComickFunDto.kt (hunk 2 above)
Comment (AntsyLich) [inline src/all/comickfun/src/eu/kanade/tachiyomi/extension/all/comickfun/ComickFunDto.kt:51]:
```suggestion
        genre = if (comic.typeGenre != null) {
            genres + comic.typeGenre
        } else {
            genres
        }
            .joinToString { it.name.trim() }
```

**Review by AntsyLich — CHANGES_REQUESTED** [review]
Give me co-author thanks

**Comment by iD-666** [comment]
Co-authored-by: @AntsyLich

**Comment by vetleledaal** [comment]
you can use https://coauthoredby.netlify.app/ , and you should add it in the commit message

**Comment by iD-666** [comment]
> Please do test your changes.

I actually did this with ChatGPT, and there is no way to run Android Studio on a smartphone 😅 :')

**Comment by alessandrojean** [comment]
Gonna mark it as draft until it gets actually tested in Android Studio. Won't be merged until that.

**Pull request closed without merging**